### ISSUE TITLE
[CI] IndicesOptionsTests.testSerialization fails reproducible on master

### ISSUE THREAD

**Issue by spinscale** [elastic/elasticsearch#30648]
While debugging another issue, this test fails reliably for me on osx and linux with this seed.

Note, this issue did happen on CI, only on my local machines, thus no link to CI.

```
./gradlew :server:test -Dtests.seed=487A3C1C76FFEBEB -Dtests.class=org.elasticsearch.action.support.IndicesOptionsTests -Dtests.method="testSerialization" -Dtests.security.manager=true -Dtests.locale=vai-Vaii -Dtests.timezone=Pacific/Marquesas
```

```
  1> [2018-05-16T02:15:38,167][INFO ][o.e.a.s.IndicesOptionsTests] [testSerialization]: before test
  1> [2018-05-16T02:15:38,383][INFO ][o.e.a.s.IndicesOptionsTests] [testSerialization]: after test
  2> REPRODUCE WITH: ./gradlew :server:test -Dtests.seed=487A3C1C76FFEBEB -Dtests.class=org.elasticsearch.action.support.IndicesOptionsTests -Dtests.method="testSerialization" -Dtests.security.manager=true -Dtests.locale=vai-Vaii -Dtests.timezone=Pacific/Marquesas
FAILURE 0.29s | IndicesOptionsTests.testSerialization <<< FAILURES!
   > Throwable #1: java.lang.AssertionError:
   > Expected: <false>
   >      but: was <true>
   >    at __randomizedtesting.SeedInfo.seed([487A3C1C76FFEBEB:1CA25D5451ADD677]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.action.support.IndicesOptionsTests.testSerialization(IndicesOptionsTests.java:51)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:564)
   >    at java.base/java.lang.Thread.run(Thread.java:844)
  2> NOTE: leaving temporary files on disk at: /Users/alr/devel/elastic/elasticsearch/elasticsearch/server/build/testrun/test/J0/temp/org.elasticsearch.action.support.IndicesOptionsTests_487A3C1C76FFEBEB-001
  2> NOTE: test params are: codec=Asserting(Lucene70), sim=RandomSimilarity(queryNorm=false): {}, locale=vai-Vaii, timezone=Pacific/Marquesas
  2> NOTE: Mac OS X 10.13.3 x86_64/Oracle Corporation 10.0.1 (64-bit)/cpus=8,threads=1,free=507352504,total=536870912
  2> NOTE: All tests run in this JVM: [IndicesOptionsTests]
Completed [1/1] in 1.33s, 1 test, 1 failure <<< FAILURES!
```

I have no real idea where to sort this under, so I picked `Core/Core`, feel free to change anytime. @javanna this might be one for you?

**Comment by spinscale** [comment]
this just happened on CI as well https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=debian/2439/console

**Comment by jpountz** [comment]
Closed via #30644.
